service/matching/task_queue_partition_manager.go

```diff
@@ -200,16 +170,17 @@ func (pm *taskQueuePartitionManagerImpl) Stop() {
 		vq.Stop()
 	}
 	pm.defaultQueue.Stop()
-	pm.goroGroup.Cancel()
+	pm.userDataManager.Stop()
 }
 
 func (pm *taskQueuePartitionManagerImpl) MarkAlive() {
 	pm.defaultQueue.MarkAlive()
 }
 
 func (pm *taskQueuePartitionManagerImpl) WaitUntilInitialized(ctx context.Context) error {
-	_, err := pm.userDataReady.Get(ctx)
+	err := pm.userDataManager.WaitUntilInitialized(ctx)
 	if err != nil {
+		pm.unloadFromEngine()
```

Comment:
why do this here? in general the pattern has been to let the thing that got the persistence error or whatever unload itself